src/neptune_mlflow_plugin/impl/plugin.py

```diff
@@ -0,0 +1,284 @@
+#
+# Copyright (c) 2023, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import uuid
+import zipfile
+from copy import deepcopy
+from datetime import datetime
+from pathlib import Path
+from typing import (
+    List,
+    Optional,
+)
+
+from mlflow.entities import (
+    DatasetInput,
+    Experiment,
+    FileInfo,
+    LifecycleStage,
+    Metric,
+    Param,
+)
+from mlflow.entities import Run as MlflowRun
+from mlflow.entities import (
+    RunData,
+    RunInfo,
+    RunStatus,
+    ViewType,
+)
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+from mlflow.store.tracking import SEARCH_MAX_RESULTS_DEFAULT
+from mlflow.store.tracking.abstract_store import AbstractStore
+from mlflow.utils.file_utils import local_file_uri_to_path
+from neptune import Run
+from neptune.attributes import FileSet
+
+from neptune_mlflow_plugin.impl.utils import (
+    encode_config,
+    parse_neptune_kwargs_from_uri,
+)
+
+
+class NeptuneTrackingStore(AbstractStore):
+    def __init__(self, store_uri: Optional[str], artifact_uri: Optional[str]):
+        self.store_uri = store_uri
+        self.artifact_uri = artifact_uri
+
+        self._neptune_kwargs = parse_neptune_kwargs_from_uri(store_uri)
+
+        self._neptune_run: Optional[Run] = None
+
+        super().__init__()
+
+        try:
+            self._login = os.getlogin()
+        except OSError:
+            # e.g. in CI
+            self._login = "runner"
+
+    def get_neptune_run(self, run_id: str) -> Run:
+        if not self._neptune_run:
+            self._neptune_run = Run(custom_run_id=run_id, **self._neptune_kwargs)
+
+        return self._neptune_run
```

Comment:
We need to be sure that `run_id` is correct + ensure that whenever we're creating a new Run, old one was stopped/synced + del.